src/NzbDrone.Core/Indexers/Definitions/Rarbg/RarbgParser.cs

```diff
@@ -23,20 +23,37 @@ public RarbgParser(IndexerCapabilities capabilities)
         public IList<ReleaseInfo> ParseResponse(IndexerResponse indexerResponse)
         {
             var results = new List<ReleaseInfo>();
+            var retryTime = 60; // Time in Seconds Default to 1 Minute
+            var responseCode = (int)indexerResponse.HttpResponse.StatusCode;
 
-            switch (indexerResponse.HttpResponse.StatusCode)
+            switch (responseCode)
             {
+                case 429:
+                    retryTime = 2 * retryTime;
+                    throw new TooManyRequestsException(indexerResponse.HttpRequest, indexerResponse.HttpResponse, retryTime);
+                case 520:
+                    retryTime = 3 * retryTime;
+                    throw new TooManyRequestsException(indexerResponse.HttpRequest, indexerResponse.HttpResponse, retryTime);
+                case 200:
+                    retryTime = 5 * retryTime;
+                    break;
                 default:
-                    if (indexerResponse.HttpResponse.StatusCode != HttpStatusCode.OK)
+                    if (responseCode != 200)
```

Comment:
Given you are handling 200 in the swtich, it will never be 200 if it ends up here.